Source/WebCore/rendering/PositionedLayoutConstraints.cpp

```diff
@@ -159,11 +159,18 @@ void PositionedLayoutConstraints::captureGridArea(const RenderBox& renderer)
         return;
 
     if (LogicalBoxAxis::Inline == m_containingAxis) {
-        m_containingRange = gridContainer->gridAreaColumnRangeForOutOfFlow(renderer);
-        m_marginPercentageBasis = m_containingRange.size();
+        auto range = gridContainer->gridAreaColumnRangeForOutOfFlow(renderer);
+        if (!range)
+            return;
+        m_containingRange = *range;
+        m_marginPercentageBasis = range->size();
     } else {
-        m_containingRange = gridContainer->gridAreaRowRangeForOutOfFlow(renderer);
-        m_marginPercentageBasis = gridContainer->gridAreaColumnRangeForOutOfFlow(renderer).size();
+        auto range = gridContainer->gridAreaRowRangeForOutOfFlow(renderer);
+        if (range)
+            m_containingRange = *range;
+        auto columnRange = gridContainer->gridAreaColumnRangeForOutOfFlow(renderer);
+        if (columnRange)
+            m_marginPercentageBasis = columnRange->size();
```

Comment:
is it okay for m_containingRange to become "stale" when range comes back as std::nullopt?

Reply:
Yeah, it will fall back to using the padding edge of the containing block (i.e. default behavior if `grid-area` is not set).

What I don't know yet is what causes us to not have this info stored in the map, if that only happens in the cases where that's the right behavior, or if it also happens in some cases where it shouldn't. That's something I still need to investigate.